src/cascadia/TerminalApp/TabBase.cpp

```diff
@@ -138,10 +138,77 @@ namespace winrt::TerminalApp::implementation
         TabViewIndex(idx);
         TabViewNumTabs(numTabs);
         _EnableCloseMenuItems();
+        _UpdateSwitchToTabKeyChord();
     }
 
     void TabBase::SetDispatch(const winrt::TerminalApp::ShortcutActionDispatch& dispatch)
     {
         _dispatch = dispatch;
     }
+
+    void TabBase::SetKeyMap(const Microsoft::Terminal::Settings::Model::KeyMapping& keymap)
+    {
+        _keymap = keymap;
+        _UpdateSwitchToTabKeyChord();
+    }
+
+    // Method Description:
+    // - Sets the key chord resulting in switch to the current tab.
+    // Updates tool tip if required
+    // Arguments:
+    // - keyChord - string representation of the key chord that switches to the current tab
+    // Return Value:
+    // - <none>
+    winrt::fire_and_forget TabBase::_UpdateSwitchToTabKeyChord()
+    {
+        SwitchToTabArgs args{ _TabViewIndex };
+        ActionAndArgs switchToTab{ ShortcutAction::SwitchToTab, args };
+        const auto keyChord = _keymap ? _keymap.GetKeyBindingForActionWithArgs(switchToTab) : nullptr;
+        const auto keyChordText = keyChord ? KeyChordSerialization::ToString(keyChord) : L"";

```

Comment:
weird enough, I think we have another keychord->string converter too living in TerminalPage. Not sure which one we should use here. This one used to be used in settings re-serialization, so it was _less_ user-facing.

The one in TerminalApp is admittedly strange. Probably best not to use it.